src/wallet/wallet.cpp

```diff
@@ -289,6 +289,27 @@ void CWallet::SetBestChain(const CBlockLocator& loc)
 {
     CWalletDB walletdb(strWalletFile);
     walletdb.WriteBestBlock(loc);
+
+    UpdateConflicted();
+}
+
+void CWallet::UpdatedBlockTip(const CBlockIndex *pindex)
+{
+    UpdateConflicted();
+}
+
+void CWallet::UpdateConflicted()
+{
+    //check for conflicted transactions
+    LOCK2(cs_main, cs_wallet);
+    for (map<uint256, CWalletTx>::iterator it = mapWallet.begin(); it != mapWallet.end(); ++it)
```

Comment:
Iterating over all wallet transactions on every block update? That seems very inefficient...